common/models/user.js

```diff
@@ -993,18 +995,22 @@ module.exports = function(User) {
 
   User.validatePassword = function(plain) {
     var err;
-    if (plain && typeof plain === 'string' && plain.length <= MAX_PASSWORD_LENGTH) {
-      return true;
+    if (!plain || typeof plain !== 'string') {
+      err = new Error(g.f('Invalid password.'));
+      err.code = 'INVALID_PASSWORD';
```

Comment:
I disagree; this isn't about telling the user (or potential attacker) that the password was _correctly formatted, but invalid_, only that the format itself is what's problematic. The only information it reveals is information they should've already had (the password should be a string!)